File: data/code_pairs/cpp/339.cpp
// stream_test.cpp
#include <fstream>
#include <cstdio>
#include <iostream>

void writeToTempFile() {
    char tempFilename[L_tmpnam];
    std::tmpnam(tempFilename); // Note: std::tmpnam has been considered dangerous; consider using std::filesystem in C++17 and beyond.
    
    std::ofstream fileStream(tempFilename, std::ios::binary);
    if (fileStream) {
        fileStream << "This is a test";
        fileStream.close();
        std::cout << "File write successful." << std::endl;
    } else {
        std::cerr << "Error occurred during file write." << std::endl;
    }
    std::remove(tempFilename); // Delete the temporary file
}

int main() {
    writeToTempFile();
    return 0;
}

File: data/code_pairs/fortran/339.f90
! stream_test.f90
program stream_test
    implicit none
    integer :: ios

    open(10, status='scratch', access='stream')
    write (10, iostat=ios) "This is a test"
    if (ios /= 0) then
        print *, "Error occurred during file write."
        stop
    endif
    close(10) ! It's a good practice to close the file
end program stream_test